doc/q3c.md

```diff
@@ -0,0 +1,121 @@
+Q3C Indexing
+============
+
+Overview          				{#q3c-overview}
+========
+
+Q3C is described by the following paper:
+
+> Koposov, S., Bartunov, O., Jul. 2006. Q3C, Quad Tree Cube –
+> The new sky-indexing concept for huge astronomical catalogues and
+> its realization for main astronomical queries (cone search and xmatch)
+> in Open Source Database PostgreSQL. In: Gabriel, C., Arviset, C.,
+> Ponz, D., Enrique, S. (Eds.), Astronomical Data Analysis Software
+> and Systems XV. Vol. 351 of Astronomical Society of the Pacific
+> Conference Series. p. 735.
+
+available online [here](http://adsabs.harvard.edu/abs/2006ASPC..351..735K).
+The authors also provide [an implementation](https://github.com/segasai/q3c).
+
+Two kinds of Q3C indexes can be produced by this library. The first kind,
+"original" Q3C indexes, are compatible with those produced by the Q3C
+PostgreSQL extension, and are provided by the
+[Q3cPixelization](\ref lsst::sphgeom::Q3cPixelization) class.
+
+The second kind, "modified" Q3C indexes, are produced according to a
+similar scheme described below. They are provided by the
+[Mq3cPixelization](\ref lsst::sphgeom::Mq3cPixelization) class.
+
+To obtain the index of unit vector P, both the original and modified
+schemes start by centrally projecting P onto the faces of the cube [-1,1]³.
+Each of the 6 cube faces is assigned a distinct face number, and a local
+coordinate system (u, v). A grid of pixels is overlaid on each cube face,
+and pixels are labeled according to a space filling curve. The index is
+obtained by combining the label of the pixel containing the face coordinates
+with the face number.
```

Comment:
If you mean, "given a point exactly on a cube edge or vertex, which face is it assigned to", I've left it as implementation defined, on the grounds that whatever choice one makes is an infinitesimal error away from a different choice. And even if I tried to match say what the Postgres extension module does (in theory), the code for computing indexes is different so that there are likely to be points very close to a cube face edge for which my implementation and the original one disagree.